File: Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        summ = []

        def addSum(node, level):
            if node is None:
                return

            if level >= len(summ):
                summ.append(node.val)
            else:
                summ[level] += node.val

            addSum(node.left, level + 1)
            addSum(node.right, level + 1)

        addSum(root, 0)
        maxx = max(summ)
        return summ.index(maxx) + 1

File: Maximum Level Sum of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> levelSum;

    void recur(TreeNode* node, int level) {
        if (!node) return;
        if (levelSum.size() <= level) {
            levelSum.push_back(0);
        }
        levelSum[level] += node->val;
        recur(node->left, level+1);
        recur(node->right, level+1);
    }

    int maxLevelSum(TreeNode* root) {
        recur(root, 0);
        int n = levelSum.size();
        int summ = levelSum[0];
        int idx = 0;
        for (int i{1}; i < n; ++i) {
            if (levelSum[i] > summ) {
                summ = levelSum[i];
                idx = i;
            }
        }
        return idx+1;
    }
};
